src/ucp/proto/proto_am.inl

```diff
@@ -349,12 +349,8 @@ ucp_proto_get_zcopy_threshold(const ucp_request_t *req,
         return max_zcopy;
     }
 
-    if (ucs_unlikely(!UCP_MEM_IS_HOST(req->send.mem_type))) {
-        return ucs_min(max_zcopy, msg_config->mem_type_zcopy_thresh[req->send.mem_type]);
-    }
-
     if (ucs_likely(UCP_DT_IS_CONTIG(req->send.datatype))) {
-        return ucs_min(max_zcopy, msg_config->zcopy_thresh[0]);
+        return ucs_min(max_zcopy, msg_config->mem_type_zcopy_thresh[req->send.mem_type]);
```

Comment:
@bureddy is this a real problem AFAIR send functions should not even check for zcopy thresh in case of generic datatype

Reply:
Yes.  

we check zcopy threshold explicitly for mem type : https://github.com/openucx/ucx/blob/master/src/ucp/tag/tag_send.c#L63

It seems zcopy threshold in case of generic datatype is setting in this function itself: https://github.com/openucx/ucx/blob/master/src/ucp/proto/proto_am.inl#L380

Reply:
adding @evgeny-leksikov 
@bureddy in case of dt_contig, zcopy_thresh=zcopy_max=rndv_thresh, for ucp_request_send_start(). so the condition ' } else if (length < zcopy_max) {' will never be met so the zcopy protocol should not be used.
did you have any test case where u see zcopy is used from dt_contig?